src/vector/indexedbd-worker.js

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// The prescribed way of doing this import would be:
+//import {IndexedDbStoreWorker} from 'matrix-js-sdk';
+// However, this still pulls in all of the js-sdk and we only use a tiny fraction
+// of it. It also causes an Olm error to appear because we don't have an Olm in scope.
+// Instead, we do this:
+import IndexedDbStoreWorker from 'matrix-js-sdk/lib/store/indexeddb-remote-worker';
```

Comment:
Sort of - the optimisation may be able to remove the unreferenced classes etc. but anything that's run in the root of the script (like the Olm detection or polyfills) can't be removed.

How about a compromise: I've moved the worker out to a separate import in src. It still needs to be imported via matrix-js-sdk/lib/indexeddb-worker (because you can only have one default) but it keeps the worker separate from everything else because it is logically separate, ie. you'd never use the worker class and the rest of the SDK in the same script. We're still importing via an import script rather than reaching right into the guts of the js-sdk though.